src/Tools/dotnet-monitor/Extensibility/ExtensionDeclaration.cs

```diff
@@ -1,42 +1,57 @@
 ﻿// Licensed to the .NET Foundation under one or more agreements.
 // The .NET Foundation licenses this file to you under the MIT license.
-// See the LICENSE file in the project root for more information.
 
-using System;
+using System.Globalization;
 
 namespace Microsoft.Diagnostics.Tools.Monitor.Extensibility
 {
     internal class ExtensionDeclaration
     {
         /// <summary>
-        /// Id of the extension. This should be namespace qualified like nuget packages.
+        /// This is the name that users specify in configuration to refer to the extension.
         /// </summary>
-        public string Id { get; set; }
+        public string Name { get; set; }
 
         /// <summary>
-        /// The version of the extension. This should be SemVer 2.0.
+        /// If specified, the executable file (without extension) to be launched when executing the extension.
         /// </summary>
-        public Version Version { get; set; }
+        /// <remarks>
+        /// Either <see cref="ExecutableFileName"/> or <see cref="AssemblyFileName"/> must be specified.
+        /// </remarks>
+        public string ExecutableFileName { get; set; }
 
         /// <summary>
-        /// This is the relative path to the executable file to be launched.
+        /// If specified, executes the extension using the shared .NET host (e.g. dotnet.exe) with the specified entry point assembly (without extension). 
         /// </summary>
-        public string Program { get; set; }
+        /// <remarks>
+        /// Either <see cref="ExecutableFileName"/> or <see cref="AssemblyFileName"/> must be specified.
+        /// </remarks>
+        public string AssemblyFileName { get; set; }
 
-        /// <summary>
-        /// This is the name that users specify in configuration to refer to the extension.
-        /// </summary>
-        public string Name { get; set; }
+        public void Validate()
+        {
+            bool hasAssemblyFileName = string.IsNullOrEmpty(AssemblyFileName);
```

Comment:
Should these be inverted?

Reply:
That's true, however the validation logic still works 😅